pkg/operator/resource/resourcemerge/object_merger.go

```diff
@@ -124,7 +142,13 @@ func MergeMap(modified *bool, existing *map[string]string, required map[string]s
 	for k, v := range required {
 		if existingV, ok := (*existing)[k]; !ok || v != existingV {
 			*modified = true
-			(*existing)[k] = v
+			// if "required" map contains a key with "-" as suffix, remove that
+			// key from the existing map instead of replacing the value
+			if strings.HasSuffix(k, "-") {
+				delete(*existing, strings.TrimRight(k, "-"))
```

Comment:
Intentional to remove arbitrary man `-` here?

Reply:
yes, you need to remove the trailing `-` to get the actual name of the field you're trying to remove:
```
original: run-level: 0
required: run-level-:
```
This is different in case of https://github.com/openshift/library-go/pull/727/files#diff-3567eb2cd10c84b30387be04408ab7b2R27 where the object being cleaned is the object that's actually getting created.

Reply:
I meant that `TrimRight` also removes `---` from `foo----`.